packages/code-studio/src/dashboard/linker/Linker.tsx

```diff
@@ -29,15 +29,85 @@ import {
 } from '../events';
 import LayoutUtils from '../../layout/LayoutUtils';
 import LinkerOverlayContent from './LinkerOverlayContent';
-import LinkerUtils from './LinkerUtils';
-import LinkType from './LinkType';
+import LinkerUtils, { Link, LinkColumn, LinkType } from './LinkerUtils';
 import { PanelManager } from '../panels';
-import { UIPropTypes } from '../../include/prop-types';
 
 const log = Log.module('Linker');
 
-class Linker extends Component {
-  constructor(props) {
+export type PanelProps = {
+  glContainer: GoldenLayout.Container;
+  glEventHub: GoldenLayout.EventEmitter;
+};
+
+export type Panel = Component<PanelProps>;
+
+export type LinkFilterMapValue = {
+  columnType: string;
+  text: string;
+  value: unknown;
+};
+
+export type LinkFilterMap = Map<string, LinkFilterMapValue>;
+
+export type LinkDataMapValue = {
+  type: string;
+  text: string;
+  value: string;
+};
+
+export type LinkDataMap = Record<string, LinkDataMapValue>;
+
+export type LinkablePanel = Panel & {
+  setFilterMap: (filterMap: LinkFilterMap) => void;
+  unsetFilterValue: () => void;
+};
+
+export function isLinkablePanel(panel: Panel): panel is LinkablePanel {
+  return (panel as LinkablePanel).setFilterMap != null;
+}
+
+interface StateProps {
+  activeTool: string;
+  isolatedLinkerPanelId?: string;
+  links: Link[];
+  timeZone: string;
+}
+
+interface OwnProps {
+  layout: GoldenLayout;
+  panelManager: PanelManager;
+  localDashboardId: string;
+}
+
+const mapState = (state: LinkerState, ownProps: OwnProps) => ({
```

Comment:
This needs an explicit return type currently since it's inferring from JS as any. So it has the right structure, but every value is any. I believe it's related to the `ConnectedProps` problem I mentioned further down in this file

Reply:
Yup adding the return type here fixes the `any` problem.